test/clj/rems/test_pdf.clj

```diff
@@ -40,6 +40,185 @@
+                        [:phrase "2001-01-01 00:00" " " "Alice Applicant lämnade in ansökan." nil nil nil]
+                        [:phrase "2003-01-01 00:00" " " "Developer begärde granskning av Carl Reviewer." nil "\nKommentar: please have a look" nil]
+                        [:phrase "2003-01-01 00:00" " " "Developer begärde beslut från användare David Decider." nil "\nKommentar: please decide" nil]
+                        [:phrase "2003-01-01 00:00" " " "David Decider behandlade ansökan." "\nDavid Decider godkände ansökan." "\nKommentar: I have decided" nil]]]]
+          only-applicant-events [[:heading {:spacing-before 20} "Händelser"]
+                                 [:list
+                                  [[:phrase "2000-01-01 00:00" " " "Alice Applicant skapade ansökan 2000/1." nil nil nil]
+                                   [:phrase "2001-01-01 00:00" " " "Alice Applicant lämnade in ansökan." nil nil nil]]]]]
+      (testing "alice should not see reviewer and decider actions"
+        (is (= [{}
+                [[:heading {:spacing-before 20} "Ansökan 2000/1: "]
+                 [:paragraph "Denna pdf skapades" " " "2010-01-01 00:00"]
+                 [:paragraph "Status" [:phrase ": " "Inlämnad"]]
+                 [:heading {:spacing-before 20} "Sökande"]
+                 [:paragraph "Sökande" ": " "Alice Applicant (alice) <alice@example.com>. Licenserna har accepterats: Ja"]
+                 []
+                 [:heading {:spacing-before 20} "Resurser"]
+                 [:list [[:phrase "Resurs" " (" "pdf-resource-ext" ")"]
+                         [:phrase "Resurs 2" " (" "pdf-resource-2-ext" ")"]]]]
+                [[:heading {:spacing-before 20} "Licenser"] []]
+                all-form-fields
+                only-applicant-events]
+               (with-language :sv
+                 (fn []
+                   (with-fixed-time (time/date-time 2010)
+                     (fn []
+                       (#'pdf/render-application (applications/get-application-for-user "alice" application-id)))))))))
+      (testing "handler should see complete application"
+        (is (= [{}
+                [[:heading {:spacing-before 20} "Ansökan 2000/1: "]
+                 [:paragraph "Denna pdf skapades" " " "2010-01-01 00:00"]
+                 [:paragraph "Status" [:phrase ": " "Inlämnad"]]
+                 [:heading {:spacing-before 20} "Sökande"]
+                 [:paragraph "Sökande" ": " "Alice Applicant (alice) <alice@example.com>. Licenserna har accepterats: Ja"]
+                 []
+                 [:heading {:spacing-before 20} "Resurser"]
+                 [:list [[:phrase "Resurs" " (" "pdf-resource-ext" ")"]
+                         [:phrase "Resurs 2" " (" "pdf-resource-2-ext" ")"]]]]
+                [[:heading {:spacing-before 20} "Licenser"] []]
+                all-form-fields
+                all-events]
+               (with-language :sv
```

Comment:
We could consider having some of these with different language.

Reply:
updated, uses all three localizations now